source/MaterialXRuntime/Private/PvtApi.cpp

```diff
@@ -173,7 +173,23 @@ void PvtApi::setupNodeImplRelationships()
         if (nodedefObj)
         {
             RtNodeDef nodedef(nodedefObj->hnd());
-            nodedef.getNodeImpls().connect(nodeimpl.getPrim());
+            RtString nodeGraphString = nodeimpl.getNodeGraph();
+            if (nodeGraphString == RtString::EMPTY)
+            {
+                nodedef.getNodeImpls().connect(nodeimpl.getPrim());
+            }
+            // Handle case where the implementation specifies the nodedef - nodegraph implementation
+            // relationship
+            else
+            {
+                PvtObject* nodeGraphObj = _nodegraphs.find(nodeGraphString);
```

Comment:
Read through the implemenetation to get the nodegraph which is the actual implementation. This means that callers don't need to about how the implementation was setup and have to do this work as it's done and cached here.